ratis-test/src/test/java/org/apache/ratis/datastream/DataStreamBaseTest.java

```diff
@@ -371,23 +397,24 @@ private void runTestDataStream(DataStreamOutputImpl out, int bufferSize, int buf
       Assert.assertEquals(sizes.get(i).longValue(), reply.getBytesWritten());
       Assert.assertEquals(reply.getType(), Type.STREAM_DATA);
     }
-
-    final RaftClientRequest header = out.getHeader();
-    for (MultiDataStreamStateMachine s : stateMachines.values()) {
-      final SingleDataStream stream = s.getSingleDataStream(header.getCallId());
-      if (stream == null) {
-        continue;
-      }
-      final RaftClientRequest writeRequest = stream.getWriteRequest();
-      if (writeRequest.getClientId().equals(header.getClientId())) {
-        Assert.assertEquals(writeRequest.getCallId(), header.getCallId());
-        Assert.assertEquals(writeRequest.getRaftGroupId(), header.getRaftGroupId());
-        Assert.assertEquals(writeRequest.getServerId(), header.getServerId());
-      }
-      Assert.assertEquals(dataSize, stream.getByteWritten());
+    try {
+      assertHeader(out.getHeader(), dataSize);
+    } catch (Throwable e) {
+      throw new CompletionException(e);
     }
   }
 
+  void assertHeader(RaftClientRequest header, int dataSize) throws Exception {
+    final Server server = servers.get(0);
```

Comment:
By doing so, this test no longer test all server's written bytes.

But I guess this is still ok, because there is code above that verifies each DATA write is successful and returned byte written matches. We have also introduced code to verify byte written on server side to make sure remote writes successful. So combined these two will still have the same test coverage.

Reply:
> By doing so, this test no longer test all server's written bytes directly.

Do you mean that it checks only the first server instead of a loop?

I found that the loop is not useful since the callId is unique.  Only the first server can match the callId.  Thus, only it will be tested.